core/types/transaction_marshalling.go

```diff
@@ -192,6 +193,29 @@ func (t *Transaction) MarshalJSON() ([]byte, error) {
 		data := tx.data()
 		enc.Data = (*hexutil.Bytes)(&data)
 		enc.To = t.To()
+	case *ArbitrumSubtypedTx:
+		subtype := uint64(tx.TxSubtype())
+		enc.Subtype = (*hexutil.Uint64)(&subtype)
+		switch subtype {
+		case ArbitrumTippingTxSubtype:
```

Comment:
alternatively we could call here a method of `ArbitrumSubtypedTx` (something like `tx.(*ArbitrumSubtypedTx).EncodeJSON(&enc)` ) and have the specific tx subtypes implement the encoding. The similar thing could be done for RLP encoding. 
This could be nice to have almost everything that's needed to implement new tx subtype in one place, but also it would take it out of context, so initially I followed geths file split.